### ISSUE TITLE
[BUG] [PHP] Symfony enum is not serialized with value like '0'

### ISSUE THREAD

**Issue by StefanFu** [OpenAPITools/openapi-generator#19562]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

enum in OpenApi yaml file with value '0' is not send. The state '1' is send.
Reason for that is a bad check in the getSerializedState() function. value '0' is false:

Generated Code:

  ```
/**
   * Gets state for serialization.
    *
    * @return string|null
    */
    public function getSerializedState(): string|null
    {
        return $this->state?->value ? (string) $this->state->value : null;
    }
```

[See PHP comparison](https://www.php.net/manual/en/types.comparisons.php)

##### openapi-generator version

Version 2.13.4

##### OpenAPI declaration file content or url

```yaml
    ItemState:
      type: string
      enum: [0, 1, 2, 3]
      x-enum-varnames:
        - OPEN
        - REVIEW
        - DONE
        - DELETED
```

##### Generation Details

`openapi-generator-cli generate -i my.yaml -g php-symfony -o src-genSrv --package-name myapi --additional-properties=phpLegacySupport=false,hideGenerationTimestamp=true",`

##### Steps to reproduce

Generate API
Via Get send the ItemState with value '0'

##### Suggest a fix

File modules/openapi-generator/src/main/resources/php-symfony/model_generic.mustache:
```
   public function getSerializedState(): string|null
    {
        return is_string($this->state?->value) ? (string) $this->state->value : null;
    }
```

**Comment by JulianVennen** [comment]
The openapi-generator version you provided seems like the version of the openapi-generator-cli, not the generator itself.
You can get the openapi-generator version by running `openapi-generator-cli version`.

It would also have been helpful if you could provide a full api spec that you can use to reproduce the issue.

I was able to reproduce the issue on 7.8.0 with this spec: https://gist.github.com/JulianVennen/9f2798ed345942e67aadd24e7c41eaa1

**Comment by wing328** [comment]
fixed via #19568 

thanks @JulianVennen for the fix
